source/features/sync-pr-commit-title.tsx

```diff
@@ -77,31 +77,24 @@ async function updateCommitTitle(): Promise<void> {
 	updateUI();
 }
 
-function disableSubmission(): void {
-	deinit();
-	getUI().remove();
-}
-
-const subscriptions: delegate.Subscription[] = [];
-
-function init(): Deinit {
-	subscriptions.push(
-		onPrCommitMessageRestore(updateUI),
-		onPrMergePanelOpen(updateCommitTitle),
-		delegate(document, '#merge_title_field', 'input', updateUI),
-		delegate(document, 'form.js-merge-pull-request', 'submit', updatePRTitle),
-		delegate(document, '.rgh-sync-pr-commit-title', 'click', disableSubmission),
-	);
-
-	return deinit;
-}
-
-function deinit(): void {
-	for (const subscription of subscriptions) {
-		subscription.destroy();
-	}
-
-	subscriptions.length = 0;
+function init(deinitSignal: AbortSignal): void {
+	const internalController = new AbortController();
+	const {signal} = internalController;
+	deinitSignal.addEventListener('abort', () => {
+		if (!signal.aborted) {
```

Comment:
Like promises, AbortControllers can only be thrown once. Successive calls are ignored, no need to check for this